actors/builtin/miner/miner_actor.go

```diff
@@ -498,25 +497,31 @@ func (a Actor) PreCommitSector(rt Runtime, params *SectorPreCommitInfo) *adt.Emp
 
 	notifyPledgeChanged(rt, newlyVested.Neg())
 
-	bf := abi.NewBitField()
-	bf.Set(uint64(params.SectorNumber))
-
-	// Request deferred Cron check for PreCommit expiry check.
-	cronPayload := CronEventPayload{
-		EventType: CronEventPreCommitExpiry,
-		Sectors:   bf,
-	}
-
+	// add precommit expiry to the queue
 	msd, ok := MaxSealDuration[params.SealProof]
 	if !ok {
 		rt.Abortf(exitcode.ErrIllegalArgument, "no max seal duration set for proof type: %d", params.SealProof)
 	}
-
 	// The +1 here is critical for the batch verification of proofs. Without it, if a proof arrived exactly on the
 	// due epoch, ProveCommitSector would accept it, then the expiry event would remove it, and then
 	// ConfirmSectorProofsValid would fail to find it.
 	expiryBound := rt.CurrEpoch() + msd + 1
-	enrollCronEvent(rt, expiryBound, &cronPayload)
+
+	// create BitField for this sector
+	bf := abi.NewBitField()
+	bf.Set(uint64(params.SectorNumber))
+
+	// Load BitField Queue for sector expiry
+	quant := st.QuantEndOfDeadline()
+	queue, err := LoadBitfieldQueue(store, st.PreCommittedSectorsExpiry, quant)
+	builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to load pre-commit expiry queue")
+
+	// add entry for this sector to the queue
+	err = queue.AddToQueue(expiryBound, bf)
+	builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to add pre-commit sector expiry to queue")
+
+	st.PreCommittedSectorsExpiry, err = queue.Root()
+	builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to save pre-commit sector queue")
```

Comment:
Please push this whole delta down to a method on miner_state, and test it directly.

Reply:
This is done.